src/fidesops/api/v1/endpoints/privacy_request_endpoints.py

```diff
@@ -569,6 +576,93 @@ def resume_privacy_request(
     return privacy_request
 
 
+def validate_manual_input(
+    manual_rows: List[Row], collection: CollectionAddress, db: Session
+) -> None:
+    """Validate manually-added data for a collection.
+
+    The specified collection must exist and all fields must be previously defined.
+    """
+    datasets = DatasetConfig.all(db=db)
+    dataset_graphs = [dataset_config.get_graph() for dataset_config in datasets]
+    dataset_graph = DatasetGraph(*dataset_graphs)
+
+    node: Optional[Node] = dataset_graph.nodes.get(collection)
+    if not node:
+        raise HTTPException(
+            status_code=HTTP_422_UNPROCESSABLE_ENTITY,
+            detail=f"Cannot save manual rows. No collection in graph with name: '{collection.value}'.",
+        )
+
+    for row in manual_rows:
+        for field_name in row:
+            if not dataset_graph.nodes[collection].contains_field(
+                lambda f: f.name == field_name  # pylint: disable=W0640
+            ):
+                raise HTTPException(
+                    status_code=HTTP_422_UNPROCESSABLE_ENTITY,
+                    detail=f"Cannot save manual rows. No '{field_name}' field defined on the '{collection.value}' collection.",
+                )
+
+
+@router.post(
+    PRIVACY_REQUEST_MANUAL_INPUT,
+    status_code=HTTP_200_OK,
+    response_model=PrivacyRequestResponse,
+    dependencies=[
+        Security(verify_oauth_client, scopes=[PRIVACY_REQUEST_CALLBACK_RESUME])
+    ],
+)
+def resume_with_manual_input(
+    privacy_request_id: str,
+    *,
+    db: Session = Depends(deps.get_db),
+    cache: FidesopsRedis = Depends(deps.get_cache),
+    manual_rows: List[Optional[Row]],
+) -> PrivacyRequestResponse:
+    """Resume a privacy request by passing in manual input for the paused collection.
+
+    If there's no manual data to submit, pass in an empty list to resume the privacy request.
+    """
+    privacy_request: PrivacyRequest = get_privacy_request_or_error(
+        db, privacy_request_id
+    )
+    if privacy_request.status != PrivacyRequestStatus.paused:
+        raise HTTPException(
+            status_code=HTTP_400_BAD_REQUEST,
+            detail=f"Invalid resume request: privacy request '{privacy_request.id}' status = {privacy_request.status.value}.",
+        )
+
+    paused_step: Optional[ActionType]
+    paused_collection: Optional[CollectionAddress]
+    paused_step, paused_collection = privacy_request.get_paused_step_and_collection()
+    if not paused_collection:
+        raise HTTPException(
+            status_code=HTTP_400_BAD_REQUEST,
+            detail=f"Cannot resume privacy request '{privacy_request.id}'; no paused collection.",
```

Comment:
Are there any valid states in the flow where a request can have a paused state, but no paused collection? If not, we might want to move the request into the errored state here.

Reply:
Yes, the privacy request can also be paused due to webhooks that run before we even build the graph. In that case, the privacy request is automatically resumed via a `reply-to` URL that we pass as a header when we call the  client-defined endpoint originally.